### PR TITLE
feat(`launch`): collapse `minLaunchTime` and `maxLaunchTime` params into atomic `LaunchTimeRange`

### LINKED ISSUE #525: Collapse `MinLaunchTime` and `MaxLaunchTime` params into single struct
As of now these two parameters are handled as separate. This also means any of the two can be updated independently.
Although in principle this is a good thing, it also prevents from making sure the two values that define the range are always consistent, *i.e.* it could be that `minLaunchTime` > `maxLaunchTime` and this would not be caught, leading to potential dangerous consequences.

We should collapse the two values into a single `range` struct that is updated as a single object. This would make possible to enforce consistency of the range, similarly to what was done for [`campaign`](https://github.com/tendermint/spn/pull/524) with `MinTotalSupply` and `MaxTotalSupply`.

### PR DESCRIPTION
**Pull request by giunatale**
Closes #525 

### What does this PR does?

The two `launch` params in `minLaunchTime` and `maxLaunchTime` are now collapsed into a single struct `LaunchTimeRange`. Genesis validation tests and localnet template genesis have also been updated accordingly.

## Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Include tests
- [x] Respect code style and lint

### CODE REVIEW

**Comment by giunatale** [comment]
wondering what caused the simulation tests to fail after merging with `develop`, any idea?

File: x/launch/types/genesis_test.go
Comment (aljo242) [inline x/launch/types/genesis_test.go:445]: Why are these tests removed?

**Review by aljo242 — CHANGES_REQUESTED** [review]

File: proto/launch/params.proto
Comment (lubtd) [inline proto/launch/params.proto:18]: What do you think about setting the type for both fields to `int64` since they represent Unix timestamp?
We should then add check for `minLaunchTime>0`

Then we can use `int64` for `remainingTime` in `MsgTriggerLaunch` and prevent from doing the `int64` conversion that can bring security problematics

**Review by lubtd — COMMENTED** [review]
👍

File: proto/launch/params.proto
Comment (giunatale) [reply proto/launch/params.proto:18]: Do we want also to allow that `MaxLaunchTime == MinLaunchTime`?

**Comment by giunatale** [comment]
> Do we want also to allow that `MaxLaunchTime == MinLaunchTime`?

I haven't allowed this in the recent changes to not change the current behavior. Holding off until I have your opinion.

Simtests fails are unrelated to this PR, will update once we have #562 merged in `develop`.

**Comment by lubtd** [comment]
>Do we want also to allow that MaxLaunchTime == MinLaunchTime?

I see no issue related to having an equal value, a single launch time would be allowed, but at the same time and don't think it's really important it's just one-second difference, in practice I expected that we will always have a range of few days of difference to give some flexibility. The choice of the minimal and maximal one matter

**Comment by giunatale** [comment]
> > Do we want also to allow that MaxLaunchTime == MinLaunchTime?
> 
> I see no issue related to having an equal value, a single launch time would be allowed, but at the same time and don't think it's really important it's just one-second difference, in practice I expected that we will always have a range of few days of difference to give some flexibility. The choice of the minimal and maximal one matter

I'll leave as it is 👍

**Comment by lubtd** [comment]
Seems like I can't serve the chain with Starport 🤔
```
starport chain serve -r
Error: failed to validate genesis state: failed to unmarshal launch genesis state: unknown field "minLaunchTime" in types.Params
```

File: x/launch/types/message_trigger_launch.go
Comment (lubtd) [inline x/launch/types/message_trigger_launch.go:12]: We should add the check `remainingTime>0` in `ValidateBasic` with a test

File: x/launch/types/params.go
Comment (lubtd) [inline x/launch/types/params.go:49]:
```suggestion
func NewParams(minLaunchTime, maxLaunchTime, revertDelay int64) Params {
```

File: x/launch/types/message_trigger_launch.go
Comment (giunatale) [reply x/launch/types/message_trigger_launch.go:12]: true! thanks for the suggestion

Comment (giunatale) [reply x/launch/types/message_trigger_launch.go:12]: true! thanks for the suggestion

Comment (giunatale) [reply x/launch/types/message_trigger_launch.go:12]: I've added a test case for `remainingTime<=0` only in the test for `ValidateBasic` following the testing conventions we have discussed 👍

**Comment by giunatale** [comment]
> Seems like I can't serve the chain with Starport 🤔
> 
> ```
> starport chain serve -r
> Error: failed to validate genesis state: failed to unmarshal launch genesis state: unknown field "minLaunchTime" in types.Params
> ```

The `config.yml` was outdated. My fault. Fixed

**Comment by lubtd** [comment]
From the `serve` issue stated above, the genesis generated from `starport serve` contains
```
"launch": {
      "chainCounter": "1",
      "chainList": [],
      "genesisAccountList": [],
      "genesisValidatorList": [],
      "params": {
        "launchTimeRange": {
          "maxLaunchTime": "604800",
          "minLaunchTime": "5"
        },
        "minLaunchTime": "5",
        "revertDelay": "5"
      },
      "requestCounterList": [],
      "requestList": [],
      "vestingAccountList": []
    },
```
`minLaunchTime` is still present in the params

**Review by lubtd — APPROVED** [review]

**Comment by aljo242** [comment]
Will approve after merge conflicts are resolved!

**Comment by giunatale** [comment]
@aljo242 can you re-review this PR?

**Review by aljo242 — APPROVED** [review]

**Comment by giunatale** [comment]
@lubtd @Pantani this needs one more review before it can be merged. Can any of you take this on? 🙏

**Review by lubtd — APPROVED** [review]

**Pull request merged**
